src/zenml/cli/cli.py

```diff
@@ -158,6 +170,15 @@ def cli() -> None:
         # packages directory
         source_utils.set_custom_source_root(source_root=os.getcwd())
 
+    # Manually show help and exit with code 0 when invoked without a subcommand.
+    # Click 8.2+ raises NoArgsIsHelpError before the callback runs when
+    # no_args_is_help=True. By relying solely on invoke_without_command=True
+    # and handling help here, we ensure consistent behavior across Click
+    # versions while leveraging our custom help formatter in ZenMLCLI.get_help().
+    if ctx.invoked_subcommand is None and not ctx.resilient_parsing:
```

Comment:
Not sure I really get why this is necessary. Why wouldn't click display the help anymore in newer versions with our previous setup?

Reply:
It's just that I can't reproduce it. I have click 8.2+, I run `zenml` or `zenml --help`, and everything shows up correct (on develop). So it uses the correct formatter I assume?